src/pal/inc/unixasmmacrosarm.inc

```diff
@@ -146,17 +147,42 @@ C_FUNC(\Name\()_End):
 
         __PWTB_StackAlloc = __PWTB_TransitionBlock
 
+	// Make the stack operation compatible with GDB and LIBUNWIND
+	PROLOG_PUSH		"{r7, lr}"
+	PROLOG_STACK_SAVE	r7
+	alloc_stack	((__PWTB_StackAlloc) + 4 * 4 + 9 * 4)
+
         .ifnc \pushArgRegs, DoNotPushArgRegs
-                PUSH_ARGUMENT_REGISTERS
+		str r0, [sp , #((__PWTB_StackAlloc) + 9 * 4)]
+		str r1, [sp , #((__PWTB_StackAlloc) + 9 * 4 + 4)]
+		str r2, [sp , #((__PWTB_StackAlloc) + 9 * 4 + 8)]
+		str r3, [sp , #((__PWTB_StackAlloc) + 9 * 4 + 12)]
```

Comment:
I would suggest renaming the PUSH/POP_ARGUMENT_REGISTERS and PUSH / POP_CALLEE_SAVED_REGISTERS to "STORE/RESTORE", change its implementation to match your changes here and to use it here for a better readability of the PROLOG_WITH_TRANSITION_BLOCK macro.
These macros are not used anywhere else, so it should be no problem.